processing/src/main/java/org/apache/druid/segment/join/table/IndexedTableJoinable.java

```diff
@@ -103,7 +103,8 @@ public ColumnValuesWithUniqueFlag getNonNullColumnValues(String columnName, fina
     }
 
     try (final IndexedTable.Reader reader = table.columnReader(columnPosition)) {
-      // Sorted set to encourage "in" filters that result from this method to do dictionary lookups in order.
+      // Use a SortedSet so InDimFilter doesn't need to create its own
+      // and to encourage "in" filters that result from this method to do dictionary lookups in order.
```

Comment:
The "encourage 'in' filters that result from this method to do dictionary lookups in order" piece isn't accurate any longer, because they no longer need encouragement. (They always use sorted sets internally.) So this is more about an optimization in creating the "in" filter itself.

Reply:
Addressed